packages/crawler/src/helpers/convert-data-helper.ts

```diff
@@ -10,16 +10,22 @@ import { getUnusedTagNames, incrementTagName } from './count-data-helper';
 import { INTERNAL_TAG_NAMES, TAG_NAMES, TagName } from '@porsche-design-system/shared';
 import { getComponentMeta } from '@porsche-design-system/component-meta';
 
-export const getPdsTagNamesWithPropertyNames = (): TagNamesWithPropertyNames =>
-  Object.values(TAG_NAMES).reduce((result, tagName) => {
-    const componentMeta = getComponentMeta(tagName);
-    return !INTERNAL_TAG_NAMES.includes(tagName)
-      ? {
-          ...result,
-          [tagName]: componentMeta.props ? Object.keys(componentMeta.props) : [],
-        }
-      : result;
-  }, {} as TagNamesWithPropertyNames);
+export const getPdsTagNamesWithPropertyNames = (): TagNamesWithPropertyNames => {
+  let result = {} as TagNamesWithPropertyNames;
```

Comment:
The previous way of implementation was the modern way, why not stick to it?

Reply:
In my opinion the reducer function is harder to read/understand, the loop is much more clearer

Reply:
ok, let's see if it helps.